Student: Duncan Ferguson <br>
Student Id: 871641260 <br>
Class: Comp 4431-1 <br>
Assignment: Excercise 2 <br>
Date: 9/24/2021 <br>
Group: Name: Broken Toe <br>
Group Members: Emma Bright, Mike Santoro <br>

In [22]:
import pandas as pd

df = pd.read_csv('ds_ex2.txt', names=["ID", "HEIGHT", "BMI", "WEIGHT", "LDL", "SBP"])  # Importing File
df.set_index("ID", inplace=True)  # Resetting the index to the ID number
df.head()

HEIGHT   BMI  WEIGHT  LDL  SBP
ID                                   
10000    1.67  19.0   52.99  115  126
10001    1.85  18.0   61.61  112   97
10002    1.80  31.0  100.44   -1  146
10003    1.49  24.0   53.28  130  116
10004    1.75  21.0   64.31  121  139

A) For the data with all records with -1 removed: <br>
-The mean of columns 2,3,4, and 5 <br>
-The correlation coefficient for each column pair <br>

In [23]:
df_a = df.copy()  # Copying the df so it can be manipulated
df_a = df_a[df_a.LDL != -1]  # removing all records with -1, note LDL is the column with -1
df_a.mean(axis=0)  # Getting the mean of columns 2,3,4,5

HEIGHT      1.682763
BMI        23.955990
WEIGHT     68.087482
LDL       129.867971
SBP       129.083130
dtype: float64

In [24]:
# The Correlation coefficient for each column pair
df_a.corr()

HEIGHT       BMI    WEIGHT       LDL       SBP
HEIGHT  1.000000 -0.034972  0.514509 -0.034972 -0.014620
BMI    -0.034972  1.000000  0.830531  1.000000  0.801063
WEIGHT  0.514509  0.830531  1.000000  0.830531  0.672957
LDL    -0.034972  1.000000  0.830531  1.000000  0.801063
SBP    -0.014620  0.801063  0.672957  0.801063  1.000000

B) For the data with all -1 values replaced by the ldl mean: <br>
-The mean of columns 2..5 <br>
-The correlation coefficient for each column pair <br>

In [28]:
df_b = df.copy()
df_b["LDL"] = df_b["LDL"].replace([-1], df_a["LDL"].mean())  # Replacing the -1 values with the column mean
df_b.mean(axis=0)

HEIGHT      1.685260
BMI        25.214000
WEIGHT     71.924640
LDL       129.867971
SBP       133.070000
dtype: float64

In [29]:
df_b.corr()

HEIGHT       BMI    WEIGHT       LDL       SBP
HEIGHT  1.000000 -0.010546  0.521627 -0.031778  0.020074
BMI    -0.010546  1.000000  0.839000  0.846905  0.825227
WEIGHT  0.521627  0.839000  1.000000  0.695751  0.709841
LDL    -0.031778  0.846905  0.695751  1.000000  0.692191
SBP     0.020074  0.825227  0.709841  0.692191  1.000000

C) Answer these questions and put your answers in the text box (but first list the names of your group members):
-For the reduced data set with the missing data records removed, which columns are correlated?
  Is this correlation strong?  Does this correlation "make sense"

For the data set with the missing data records removed we define correlation as values that are above 0.8.
Meaning that we found that we found the following columns to be strongly correlated.
[BMI and Weight] This correlation makes sense since BMI is dependent on weight.
[BMI and SBP] Weight loss significantly reduces blood pressure. Which means that this correlation makes sense.
[WEIGHT and LDL] Weight loss helps lower LDL cholesterol, making this correlation make sense
[LDL and BMI] Both are contributing risk factors for cardiovascular disease in obese individuals
[LDL and SBP] LDL is a measure of the "bad cholesterol" in your arteries which would make sense that with the build up
there would be an increase in blood pressure.

-Did replacing the "-1" values with the ldl mean approach change the mean and correlation values
compared to omitting the records?  How did they change?

Yes, replacing the -1 values with the ldl mean approach did change the mean and correlation values
compared to omitting the records.

The Height mean had a negligible change going from 1.682763 to 1.685260
The BMI mean increased from  23.955990 to 25.214000
The Weight mean increased from 68.087482 to 71.924640
The SBP mean had the greatest increase going from 129.083130 to 133.070000

For the correlations:
There were noticeable differences in correlation between
[SBP, LDL] Changing from a strong correlation of .801603 to a much weaker correlation of .692191
[LDL, BMI] Changing from a correlation of 1 to .846905

-Which method do you think is "better" and why?
I believe taking removing the rows where LDL has values of -1 is the proper way to go.
We're removing about a 1/5 of the data going from 500 to 409 rows. Not the largest loss of data.
When we did the mean replacement there were significant changes in the correlations between LDL with SBP and BMI.
Both of those correlations made sense, so when we do the mean replacement it suggests that we're skewing the
data. Additionally, because we are looking at the correlations, I believe it is best not to mock any data.

Group Members: Emma Bright and Mike Santoro <br>
